recipes/boost/all/conanfile.py

```diff
@@ -225,6 +229,12 @@ def _python_executable(self):
         exe = self.options.python_executable if self.options.python_executable else sys.executable
         return str(exe).replace('\\', '/')
 
+    @property
+    def _is_windows_platform(self):
+        return ((self.settings.os == "Windows") and
+                (self.settings.os.subsystem in [None, "cygwin"]) or
+                self.settings.os in ["WindowsStore", "WindowsCE"])
```

Comment:
Does the subsystem matter?
Isn't checking for `settings.os` enough?
```suggestion
        return self.settings.os in ("Windows", "WindowsStore", "WindowsCE")
```
If something is built for Windows, isn't that enough?

Reply:
For the wsl subsystem I would say it matters. As this is really a Linux (WM), I would expect boost to not treat it as a windows system.

As for the other subsystems, I'm not that familiar with them. But the check in #1704 did treat cygwin as being non posix (`_is_posix_platform` check there). And it is also handled special by the boost build system:
https://github.com/boostorg/locale/blob/ccb8fbb9a1a0dbdffb1054ffa34e4aba1e425642/build/Jamfile.v2#L271-L275

Reply:
I don't know much about wsl, but aren't applications built with wsl actually Linux applications (os=Windows)?

Reply:
Yes, applications build with in the wsl should be just Linux applications. Was also surprised that the wsl did show up in that list.

I just did install conan under the wsl, and for me the default config of conan was `os=Linux`. Maybe someone else can explain for what `os=Windows; os.subsystem=wsl` is actually used, and how it should be treated.

Reply:
WSL is capable of running:
- most of regular Linux executables (ELF files) which use Linux syscalls
- regular Windows executable (PE files)

the first case is definitely for `os=Linux` (as these compiled ELF files will be run on regular Linux, as well on WSL/WSL2)
the second is probably `os=Windows; os.subsystem=wsl` if you want to use build or use Windows applications via WSL layer (e.g. launching bash and other tools from WSL instead of MSYS)

Reply:
Thanks for the explanation. I just did a test with `-s os.subsystem=wsl` and both `iconv`/`icu` disabled, and boost did not complaint.
But I also did not really see a difference in the output (e.g. paths there still `C:\Users\..`, and not something like `/mnt/c/...`). So unsure if I really tested it correctly.

So adding `wsl` to the subsystems here seems form my current understanding to be the correct thing (but I'm still not 100% sure).

Reply:
subsystem affects a couple of functions, such as:
- unix_path
- run_in_windows_bash

if recipe doesn't use any of them, there will be no difference in output
as I remember, boost build system doesn't rely on bash, so subsystem might be completely irrelevant for the package
(except the fact you are using the correct compiler targeting the correct subsystem, but that's out of our control right now)

Reply:
We had done some experiments with some of our testers, my notes match SSE4's comment https://github.com/conan-io/conan-center-index/pull/5553/files#r634440165. It makes the executable compatible with both environments.

I suspect this mode of working is not supported by Boost, https://www.boost.org/doc/libs/1_76_0/more/getting_started/windows.html notes mingw is not support which (to my knowledge) is similar. 

I have no objections but I share the sentiment that perhaps if best to leave out the subsystem. Since it introduces more complexity (and Boost is already huge).

Reply:
removed the check for the subsystem